### ISSUE TITLE
"Short" and "Long" can be misleading (SC 1.1.1)

### ISSUE THREAD

**Issue by MakotoUeki** [w3c/wcag#2117]
I created this issue on behalf of the [Alt Text Subgroup](https://www.w3.org/WAI/GL/task-forces/silver/wiki/Alt_Text_Subgroup ) of the Silver Task Force. 

### Explanation:
In the "[Sufficient Techniques" section in “Understanding WCAG 2.1](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content#techniques),” the following terms can be confusing, especially for those who are new to web (digital) accessibility.

- short description
- short text alternative
- long text alternative
- long description

How can we determine if the text alternative is "short" or "long" when we have not specified a character limit for the value of the alt attribute?

Also, "short" often leads people to think that the short text alternative can meet the SC 1.1.1 even if it is not equivalent to the non-text content. They tend to mistakenly conclude that "short" is more important  than "equivalent."

#### Proposed changes:

- "short description" to "description"
- "short text alternative" to "text alternative"
- "long text alternative" to "detailed text alternative"
- "long description" to "detailed description"

#### Scope
"Sufficient Techniques" section in "Understanding WCAG 2.1"
Titles for the Techniques and Failures in “Techniques for WCAG 2.1"

#### Examples of Techniques/Failures
Here are some examples of pages that use the terms "short text alternative," "long text alternative," etc.

- [G73: Providing a long description in another location with a link to it that is immediately adjacent to the non-text content](https://www.w3.org/WAI/WCAG21/Techniques/general/G73)
- [G74: Providing a long description in text near the non-text content, with a reference to the location of the long description in the short description](https://www.w3.org/WAI/WCAG21/Techniques/general/G74)
- [G92: Providing long description for non-text content that serves the same purpose and presents the same information](https://www.w3.org/WAI/WCAG21/Techniques/general/G92)
- [F67: Failure of Success Criterion 1.1.1 and 1.2.1 due to providing long descriptions for non-text content that does not serve the same purpose or does not present the same information](https://www.w3.org/WAI/WCAG21/Techniques/general/F67)

**Comment by patrickhlauke** [comment]
i'd be careful not to enshrine some arbitrary character count as a hard "anything below this number is 'short', anything above this number is 'long'", as that's also language-dependent (e.g. thinking CJK where a single character/glyph represents a whole word/concept, so a 'short' text may still end up being incredibly dense compared to a language like, say, German)

**Comment by MakotoUeki** [comment]
@mraccess77 @bruce-usab
Thank you so much for discussing the character limit. We (Alt Text subgroup) are going to create a new issue about it. I'll share it with you once the issue is created.

@patrickhlauke
Agreed. WCAG is an international guideline. We should not make it JAWS-based and English-based.

**Comment by GreggVan** [comment]
Two thoughts

1) Perhaps use Word vs characters? 

These are good rules of thumb - for English  - but may not be for other languages.  
- Maybe we should use word count instead of characters? 

Another vote for words is the following.  Which is better?

 -  Emmanuel being dragged away from the dungeon by her great-grandmother.
 -  She is moved away from it by her.  

This is pretty extreme example but — sometimes long words convey information better but increase character count  (also a problem with "simple language" metrics that use syllable counts). 

A bigger problem is languages where the characters per word vary greatly. 


2)  Perhaps recommend short summary then  any IMPORTANT detail - vs  short. 

  I think it is better to convey all the important information in an image rather than worry about character count.   For example — if there is a diagram in a powerpoint (or other presentation slide)  it is better to convey all of the IMPORTANT information int the diagram than it is to say something like  "diagram showing the relative sales by sector for the 15  most common sales strategies"   or   "critical stress points in our product"    which only lets the person know what it is that everyone else knows but they do not. 

I often find that I have to create long alt texts for images in order to provide individuals who are blind with all of the same information as everyone else.  I always start with a short summative description - followed  by the details.   They can always move off of the picture if they are not interested in the details (like sighted persons can look away if they are not interested in details ) but not including the details in order to keep the description short —  does not seem to me to be making it more accessible. 

Best 

gregg

**Comment by mraccess77** [comment]
@GreggVan wrote ") but not including the details in order to keep the description short —  does not seem to me to be making it more accessible. "  

I agree.  My point is that longer descriptions are easier to parse in a form where there are line breaks, paragraphs, lists, headings, tables, etc. rather than just as a string of characters.

My mention of character count was not actually to specify a certain number of characters but to attempt to get at a level of detail word, sentences, etc. where there is a split between something crammed into one attribute versus more structured content.  It was not meant to encourage people to use only short descriptions where a long one was needed or to force a certain length.

**Comment by MakotoUeki** [comment]
@GreggVan Thank you very much for your comments. I'm doing "short summary then any IMPORTANT detail" all the time. So I do agree. 

I've seen so many cases where the alt attribute value is ONLY short summary and it is not alternative for an image. That's why we think the technique title saying "short" is a problem.

@mraccess77 @bruce-usab @patrickhlauke @GreggVan 
I'd like to hear from you all about what do you think about our proposed changes. The character limit will be discussed in the other issue we will create soon :)

> **Proposed changes:**
> - "short description" to "description"
> - "short text alternative" to "text alternative"
> - "long text alternative" to "detailed text alternative"
> - "long description" to "detailed description"

**Comment by bruce-usab** [comment]
My recommendation is that the top-level normative phrasing align with the well travelled terms in 2.x 1.11.  That is:
*  a text alternative that provides descriptive identification
*  a text alternative that serves the equivalent purpose

I am of the opinion that adjectives like short/long/detailed are less helpful.

**Comment by MakotoUeki** [comment]
@bruce-usab Let me check if we are on the same page.

What we'd like to propose is to change the words, "short" and "long", in the "Sufficient Techniques" section in “Understanding WCAG 2.1. Because these wording are too ambiguous and confusing especially for those who are new to web accessibility.

- G94: Providing **short** text alternative for non-text content that serves the same purpose and presents the same information as the non-text content
- G95: Providing **short** text alternatives that provide a brief description of the non-text content using one of the following **Short** text alternative techniques for Situation B AND one of the following **Long** text alternative techniques for Situation B :

Do you mean you are okay with the current wording? If no, how would you change these technique titles?

For instance, G94 title often misleads people to write short alt text (e.g. "Photo", "Chart", "People") which is not equivalent for an image. 

In G95, what is "short" and what is "long" in the first place?

These are the issues we'd like to solve here.

**Comment by bruce-usab** [comment]
@MakotoUeki (et al.) &ndash; I completely agree that <q>short</q> and <q>long</q> have proven to be problematic.  I agree that these words are too ambiguous and confusing, especially for those who are new to web accessibility.

I do not think the difficulty can be fixed merely by deleting the word <q>short</q> and replacing <q>long</q> with <q>descriptive</q>.

> Do you mean you are okay with the current wording?

Well, yes, I am _okay_ with the current wording, because it is a document that achieved WG consensus, and I gave my assent then, and even now I am not finding anything _wrong_.  That said, I am all for working to improve it!

> If no, how would you change these technique titles?

I agree with taking <q>short</q> out of these technique titles.  I think the more important thing is to rewrite throughout so that providing <q>descriptive identification</q> is the main idea, not a <q>brief description</q>.
